### DISCUSSION TITLE
crLud? I'd like to see support for a separate LIST permission

### DISCUSSION THREAD

**Discussion by SeanDylanGoff** [directus/directus#11464] [Feature Requests]
It seems to me that quite a common use-case is to for example have photos attached to articles which should be public, while still leaving the other files private. The way this is often done is by having UUIDs as PKs (which directus supports), and only allowing public ReadOne and not ReadMany.

Thus separating Read into Read and List/ReadMany would be great and shouldn't increase complexity too much (i hope :D )

**Comment by rijkvanzanten** [comment]
Agreed! 

We should find a better acronym than CrLud though, and add an S for "share". Maybe CUDdLeRS, CURSeDLy, SCRUpLeD?! 😄

**Reply by xamiam** [reply]
Is there any other way to block listing all files via API but still have access to single files using their hash-id? 🙏🏼

**Reply by rijkvanzanten** [reply]
Not without some custom hook/endpoint magic, no

**Reply by JonL1** [reply]
LUDCRS seems fine as directus is software out of this world. You could even say it's ludicrous.

**Comment by loan-talvat** [comment]
To restrict the list of files, is it not possible to simply make an environment variable? Do not list but be able to access files if they are called on /assets with their token for public role

It's a security concern, or a loss of functionality (sharp.js...) with a custom endpoint

**Reply by rijkvanzanten** [reply]
I don't think there should be a difference in files vs other data sets personally 🤔

**Reply by xamiam** [reply]
Exactly. Individual entries could also be accessed via a hash while the list may be hidden. It's a nobrainer. The strength of the hash could also be defined in Directus or simply chosen by the user.

**Reply by loan-talvat** [reply]
> I don't think there should be a difference in files vs other data sets personally 🤔 

I have no security breach on the other data because the permissions filters are enough 


On the other hand, the files are not only given in the database.. these are confidential documents in our case


The token to retrieve them is sufficient security, it's really really the listing that is the problem 


The only solution found to date is to make a "public" folder and to filter in the API, for confidential data to be accessible outside of directus, I made a role and a user who has the necessary permissions. but it's more DIY than clean development, it's a shame and a bit frustrating 😵‍💫😅

(So sorry for my English, say thx to Google translate aha)

**Comment by licitdev** [comment]
Here's an unofficial workaround. An extension to disable listing of routes and collections. 
https://www.npmjs.com/package/directus-extension-disable-listing

> **Warning**
> The functionality may break in the future as it overrides the internal routing stack in `express`. 
> Kindly proceed with caution.

**Reply by wuservices** [reply]
Would it be possible to do something similar with a filter hook extension?

For example, throwing an error when trying to list files for a public user, or a specific public folder since external users shouldn't be able to list either if they authenticate.

If so, would that be less brittle in case of API route or Express changes?

**Comment by abdonrd** [comment]
I like this proposal! I think it would be good.

**Reply by bheywar** [reply]
Hello, did you mention me here?

**Comment by madc** [comment]
@rijkvanzanten Is there any progress on this concept? I think the "share" permission is now already used otherwise, right?

**Reply by rijkvanzanten** [reply]
No progress yet. It makes total sense to me for things like `/files`, but I'm wondering about the configuration overhead of configuring both (we'd need some sort of "use permissions from X" toggle to make sure you don't have to reconfigure it every time)

>  I think the "share" permission is now already used otherwise, right?

Yup! The share permission is used to control what you are allowed to share 👍🏻

**Comment by rijkvanzanten** [comment]
Heya! Thanks for opening this feature request!

This feature request has received over **15 votes** from the community. This means we'll move this feature request to the Under Review state! 

The Core team will schedule a meeting to review this request as soon as possible. The discussion will then be approved or denied. You may or may not be invited to join this meeting with the core team.

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._

**Comment by pwFoo** [comment]
Searching for the feature to set permissions to files based on the collection it is used with instead of global files read permission to everyone... Following here...

**Reply by rijkvanzanten** [reply]
I don't think the feature request in this thread is what you're talking about. Make sure to upvote https://github.com/directus/directus/discussions/3144 tho!

**Comment by rijkvanzanten** [comment]
A new feature request came in about the same thing in the new rfc format here: https://github.com/directus/directus/discussions/24075?notification_referrer_id=NT_kwDOAIt7GbMxMzQ4MjExMTc4NTo5MTQxMDE3

**Comment by neobarthold** [comment]
Is this planned in any way?
